static/app/views/organizationContext.tsx

```diff
@@ -1,5 +1,147 @@
+  static childContextTypes = {
+    organization: SentryTypes.Organization,
+  };
+
+  getChildContext() {
+    return {organization: this.props.value};
+  }
+
+  render() {
+    return this.props.children;
+  }
+}
+/**
+ * Context provider responsible for loading the organization into the
+ * OrganizationStore if it is not already present.
+ */
+export function OrganizationContextProvider({children}: Props) {
+  const api = useApi();
+  const configStore = useLegacyStore(ConfigStore);
+
+  const {organizations} = useLegacyStore(OrganizationsStore);
+  const {organization, error} = useLegacyStore(OrganizationStore);
+
+  const hasMadeFirstFetch = useRef(false);
+
+  const lastOrganization: string | null =
+    configStore.lastOrganization ?? organizations[0]?.slug ?? null;
+
+  const routes = useRoutes();
+  const params = useParams<{orgId?: string}>();
+
+  // Determine orgSlug from the parameters
+  const orgSlug = params.orgId || lastOrganization;
+
+  const handleLoad = useCallback(() => {
+    if (!orgSlug) {
+      return;
+    }
+
+    metric.mark({name: 'organization-details-fetch-start'});
+    fetchOrganizationDetails(api, orgSlug, false, hasMadeFirstFetch.current);
+    hasMadeFirstFetch.current = true;
+  }, [api, orgSlug]);
+
+  // If the organization slug differs from what we have in the organization
+  // store reload the store
+  useEffect(() => {
+    // Nothing to do if we already have the organization loaded
+    if (organization && organization.slug === orgSlug) {
+      return;
+    }
+
+    handleLoad();
+  }, [orgSlug, organization, handleLoad]);
+
+  // Take a measurement for when organization details are done loading and the
+  // new state is applied
+  useEffect(
+    () => {
+      if (organization === null) {
+        return;
+      }
+
+      metric.measure({
+        name: 'app.component.perf',
+        start: 'organization-details-fetch-start',
+        data: {
+          name: 'org-details',
+          route: getRouteStringFromRoutes(routes),
+          organization_id: parseInt(organization.id, 10),
+        },
+      });
+    },
+    // Ignore the `routes` dependency for the metrics measurement
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [organization]
+  );
+
+  // Configure sentry SDK scope to have organization tag
+  useEffect(() => {
+    if (organization === null || error) {
+      return;
+    }
+
+    Sentry.configureScope(scope => {
+      // XXX(dcramer): this is duplicated in sdk.py on the backend
+      scope.setTag('organization', organization.id);
+      scope.setTag('organization.slug', organization.slug);
+      scope.setContext('organization', {id: organization.id, slug: organization.slug});
+    });
+  }, [organization, error]);
+
+  const {user} = configStore;
+
+  // If we've had an error it may be possible for the user to use the sudo
+  // modal to load the organization.
+  useEffect(() => {
```

Comment:
I would colocate this logic with the request inside handleLoad as it's related to what that returns, easier to just keep it all in one place.

Reply:
Unfortunately because of the way our fetching works it's difficult to colocate.